integration/nevermined/SecondaryMarket.test.ts

```diff
@@ -0,0 +1,505 @@
+                    )
+                }
+            })
+
+            it('As collector1 I setup an agreement for selling my NFT', async () => {
+                const result = await nftSalesTemplate.createAgreementFromDDO(
+                    agreementId2,
+                    ddo,
+                    assetRewards2,
+                    collector2,
+                    numberNFTs2,
+                    collector1,
+                    collector1
+                )
+                assert.isTrue(result)
+
+                const status = await nftSalesTemplate.getAgreementStatus(agreementId2)
+                assert.equal(
+                    status && status.lockPayment.state,
+                    ConditionState.Unfulfilled
+                )
+                assert.equal(
+                    status && status.transferNFT.state,
+                    ConditionState.Unfulfilled
+                )
+                assert.equal(
+                    status && status.escrowPayment.state,
+                    ConditionState.Unfulfilled
+                )
+            })
+
+            it('As collector2 I am locking the payment', async () => {
+                await collector2.requestTokens(nftPrice2 / scale)
+
+                const collector2BalanceBefore = await token.balanceOf(collector2.getId())
+                assert.equal(
+                    collector2BalanceBefore,
+                    initialBalances.collector2 + nftPrice2
+                )
+
+                const receipt = await nevermined.agreements.conditions.lockPayment(
+                    agreementId2,
+                    ddo.id,
+                    assetRewards2.getAmounts(),
+                    assetRewards2.getReceivers(),
+                    token.getAddress(),
+                    collector2
+                )
+                assert.isTrue(receipt)
+
+                const collector2BalanceAfter = await token.balanceOf(collector2.getId())
+                const escrowPaymentConditionBalance = await token.balanceOf(
+                    escrowPaymentCondition.getAddress()
+                )
+                assert.equal(collector2BalanceAfter - initialBalances.collector2, 0)
+                assert.equal(
+                    escrowPaymentConditionBalance -
+                        initialBalances.escrowPaymentCondition,
+                    nftPrice2
+                )
+            })
+
+            it('As collector1 I can check the payment and transfer the NFT to collector2', async () => {
+                const nftBalanceCollector1Before = await nftUpgradeable.balance(
+                    collector1.getId(),
+                    ddo.id
+                )
+                const nftBalanceCollector2Before = await nftUpgradeable.balance(
+                    collector2.getId(),
+                    ddo.id
+                )
+
+                const receipt = await nevermined.agreements.conditions.transferNft(
```

Comment:
Also, here the right to transfer should be supported by the "delegate to gateway" flow that we have for normal first-hand transfers?